### PR TITLE
Add gprimeltecan, fortuna3g, fortunaltezt

### PR DESCRIPTION
**Pull request by stephan-gh**
This is a minor cleanup of #227. The main change is that more hardware is placed in the common `gprime.dtsi`, e.g. the sensors, parts of NFC and the pwm-backlight. The hardware is then selected in the variant-specific dts parts as needed.

@wonderfulShrineMaidenOfParadise, @SiddharthManthan, @buddyjojo Please check that the changes I made look good to you and that **it works properly**, since I might have messed something up.

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gprime.dtsi [hunk 1]
```diff
@@ -74,6 +87,29 @@
...
+		touchscreen-size-x = <540>;
+		touchscreen-size-y = <960>;
+
+		vcca-supply = <&reg_vdd_tsp_a>;
+		vdd-supply = <&pm8916_l6>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&tsp_int_default>;
+
+		linux,keycodes = <KEY_APPSELECT KEY_BACK>;
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-gprime.dtsi:239]: does the multi-touch work?

File: arch/arm64/boot/dts/qcom/Makefile [hunk 2]
```diff
@@ -28,7 +28,6 @@ dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-a3u-eur.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-a5-zt.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-a5u-eur.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-e7.dtb
-dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-samsung-gprime.dtb
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/Makefile:31]: This commit drops the dtb and doesn't actually introduce anything. Is it supposed to be squashed with some other commits maybe? I'd assume that an introductory commit for similar devices would look like this: https://github.com/msm8916-mainline/linux/commit/45f778028c29d144ff27e86aa63bb49b1b2cf418

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gprimeltecan.dts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	chassis-type = "handset";
+
+	reserved-memory {
+		/* Additional memory used by Samsung firmware modifications */
+		tz-apps@85500000 {
+			reg = <0x0 0x85500000 0x0 0xb00000>;
+			no-map;
+		};
+
+		mpss_mem: mpss@86800000 {
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-gprimeltecan.dts:19]: I'd then prefer to have the modem mem addition in the modem commits and all other stuff (model and tz_apps) in the initial commit maybe?

File: arch/arm64/boot/dts/qcom/msm8916-samsung-fortuna3g.dts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	chassis-type = "handset";
+
+	reserved-memory {
+		/* Additional memory used by Samsung firmware modifications */
+		tz-apps@85a00000 {
+			reg = <0x0 0x85a00000 0x0 0x600000>;
+			no-map;
+		};
+
+		mpss_mem: mpss@86800000 {
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-fortuna3g.dts:19]: same as the gprimeltecan

File: arch/arm64/boot/dts/qcom/msm8916-samsung-fortunaltezt.dts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	chassis-type = "handset";
+
+	reserved-memory {
+		/* Additional memory used by Samsung firmware modifications */
+		tz-apps@85a00000 {
+			reg = <0x0 0x85a00000 0x0 0x600000>;
+			no-map;
+		};
+
+		mpss_mem: mpss@86800000 {
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-fortunaltezt.dts:31]: same as other two

File: arch/arm/boot/dts/qcom-msm8916-samsung-fortunaltezt.dts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include "arm64/qcom/msm8916-samsung-fortunaltezt.dts"
+#include "qcom-msm8916-smp.dtsi"
+
+&tsens {
+	qcom,srot-locked;
```
Comment (TravMurav) [inline arch/arm/boot/dts/qcom-msm8916-samsung-fortunaltezt.dts:6]: maybe have a comment here?

File: arch/arm64/boot/dts/qcom/Makefile (hunk 2 above)
Comment (wonderfulShrineMaidenOfParadise) [reply arch/arm64/boot/dts/qcom/Makefile:31]: https://github.com/msm8916-mainline/linux/pull/228/commits/00c01dcc6f589fe75eff94fa03b14a27a3e2174b#diff-6ef2edc037b112b898a8ae8cf2e368a6cbc75dda5cc8a5dc9be6b1146ca4569dR32
https://github.com/msm8916-mainline/linux/pull/228/commits/2661127949f6f37c1c1e1a8b3025c58a28137b81#diff-6ef2edc037b112b898a8ae8cf2e368a6cbc75dda5cc8a5dc9be6b1146ca4569dR31
https://github.com/msm8916-mainline/linux/pull/228/commits/b867e93b06e82bb473aeb502d8326a9fb7b1faaf#diff-6ef2edc037b112b898a8ae8cf2e368a6cbc75dda5cc8a5dc9be6b1146ca4569dR31

New dtb will be added separately in 3 init commits.

File: arch/arm64/boot/dts/qcom/msm8916-samsung-gprimeltecan.dts (hunk 3 above)
Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-gprimeltecan.dts:19]: I think we might be able to upstream modem stuff soon so I'd keep it in one commit for now.

File: arch/arm64/boot/dts/qcom/Makefile (hunk 2 above)
Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/Makefile:31]: I'll squash this one probably somehow but for authorship I think it's better to add the dtbs separately.

**Review by TravMurav — APPROVED** [review]
assuming that the commits are properly squashed on merge/rebase, it looks good to me

**Comment by wonderfulShrineMaidenOfParadise** [comment]
Looks good after testing on fortunaltezt.

**Comment by SiddharthManthan** [comment]
Works well on fortuna3g

**Comment by stephan-gh** [comment]
Pushed to rebase

**Pull request closed without merging**